File: 架构/code/DataBindingTest/app/src/main/java/com/wz/databindingtest/viewmodel/MainViewModel.java
package com.wz.databindingtest.viewmodel;

import android.util.Log;
import android.view.View;

import androidx.databinding.ObservableBoolean;
import androidx.lifecycle.ViewModel;

/**
 * create by wangzhen 2022/4/1
 */
public class MainViewModel extends ViewModel {

    public ObservableBoolean check = new ObservableBoolean(false);


    public void onCheckChange() {
        Log.d("TAG", "onCheckChange");
    }

    public void onMyOnCheckChange(View view, boolean check) {
        Log.d("TAG", "onMyOnCheckChange check = " + check);
    }

}

File: Kotlin/code/Coroutine/app/src/main/java/com/example/coroutine/viewmodel/MainViewModel.kt
package com.example.coroutine.viewmodel

import android.view.ViewManager
import androidx.lifecycle.MutableLiveData
import androidx.lifecycle.ViewModel
import androidx.lifecycle.viewModelScope
import com.example.coroutine.api.Todo
import com.example.coroutine.repository.TodoRepository
import kotlinx.coroutines.launch

/**
 *
 * create by wangzhen 2021/12/23
 */
class MainViewModel : ViewModel() {

    val todoLiveData = MutableLiveData<Todo>()

    private val todoRepository = TodoRepository()

    fun getTodo(id: Int) {
        viewModelScope.launch {
            todoLiveData.value = todoRepository.getTodo(id)
        }
    }

}